src/openforms/submissions/logic/actions.py

```diff
@@ -191,6 +191,42 @@ def get_action_log_data(
         }
 
 
+@dataclass
+class StepApplicableAction(ActionOperation):
+    form_step_identifier: str
+
+    @classmethod
+    def from_action(cls, action: ActionDict) -> "StepApplicableAction":
+        return cls(
+            form_step_identifier=action["form_step_uuid"],
+        )
+
+    def apply(
+        self, step: SubmissionStep, configuration: FormioConfigurationWrapper
+    ) -> None:
+        execution_state = (
+            step.submission.load_execution_state()
+        )  # typically cached already
+        submission_step_to_modify = execution_state.resolve_step(
+            self.form_step_identifier
+        )
+        submission_step_to_modify.is_applicable = True
+
+    def get_action_log_data(
+        self,
+        component_map: Dict[str, ComponentMeta],
+        all_variables: Dict[str, FormVariable],
+        initial_data: dict,
+        log_data: dict[str, JSONValue],
+    ) -> JSONObject:
+        return {
+            "type_display": LogicActionTypes.get_label(
+                LogicActionTypes.step_applicable
+            ),
+            "step_name": self.form_step_identifier,
+        }
```

Comment:
I see from codecov that this is not covered by tests?

Reply:
Saw that too, but iirc the other `get_action_log_data` calls aren't either

Reply:
the logic logging is up for rework, but I'll only have a meeting about that with DH in a couple of weeks. We do know that the current state of logging is untrustworthy, so okay to skip tests here too.

Related ticket: #3239 and that probably links to a bunch of other tickets.